pkg/provider/cloud/kubevirt/provider.go

```diff
@@ -155,8 +177,12 @@ func (k *kubevirt) CleanUpCloudProvider(ctx context.Context, cluster *kubermatic
 		return cluster, fmt.Errorf("failed to delete namespace %s: %w", cluster.Status.NamespaceName, err)
 	}
 
+	if err := deleteDataVolumeClonerRoleBinding(ctx, fmt.Sprintf("%s-%s", dataVolumeRoleBinding, cluster.Status.NamespaceName), client); err != nil && !apierrors.IsNotFound(err) {
+		return cluster, fmt.Errorf("failed to delete rolebinding %s: %w", fmt.Sprintf("%s-%s", dataVolumeRoleBinding, cluster.Status.NamespaceName), err)
+	}
+
 	return update(ctx, cluster.Name, func(updatedCluster *kubermaticv1.Cluster) {
-		kuberneteshelper.RemoveFinalizer(updatedCluster, FinalizerNamespace)
+		kuberneteshelper.RemoveFinalizer(updatedCluster, FinalizerNamespace, FinalizerClonerRoleBinding)
```

Comment:
We should not remove the 2 finalizers at the same time.
Let's remove `FinalizerNamespace` when the namespace is succesfully deleted, and let's remove `FinalizerClonerRoleBinding` when the RoleBinding is deleted.
They might not succeed in the same loop and their presence will be linked to exactly to remains to finalize.